src/DotVVM.Framework/Compilation/Javascript/JavascriptTranslatableMethodCollection.cs

```diff
@@ -116,11 +116,23 @@ public void AddDefaultMethodTranslators()
             AddPropertyGetterTranslator(typeof(string), nameof(string.Length), lengthMethod);
             AddMethodTranslator(typeof(Enums), "GetNames", new EnumGetNamesMethodTranslator(), 0);
 
-            JsExpression indexer(JsExpression[] args, MethodInfo method) =>
+            JsExpression listIndexer(JsExpression[] args, MethodInfo method) =>
                 BuildIndexer(args[0], args[1], method.DeclaringType.GetProperty("Item"));
-            AddMethodTranslator(typeof(IList), "get_Item", new GenericMethodCompiler(indexer));
-            AddMethodTranslator(typeof(IList<>), "get_Item", new GenericMethodCompiler(indexer));
-            AddMethodTranslator(typeof(List<>), "get_Item", new GenericMethodCompiler(indexer));
+            JsExpression dictionaryGetIndexer(JsExpression[] args, MethodInfo method) =>
+                new JsIdentifierExpression("dotvvm").Member("dictionaryHelper").Member("getItem").Invoke(args[0], args[1]);
+            JsExpression dictionarySetIndexer(JsExpression[] args, MethodInfo method) =>
+                new JsIdentifierExpression("dotvvm").Member("dictionaryHelper").Member("setItem").Invoke(args[0].Clone(), args[1], args[2], args[0].EnsureObservableWrapped());
```

Comment:
I'm not sure if this wouldn't better on cases like: `(C ? Dictionary1 : Dictionary2)["ss"] = X`

```suggestion
                new JsIdentifierExpression("dotvvm").Member("dictionaryHelper").Member("setItem").Invoke(args[0].Clone(), args[1], args[2], args[0].WithAnnotation<ResultShouldBeObservable>());
```

Reply:
I just tested original `setItem` implementation and the conditional expression works as expected in both cases (there is no `ResultShouldBeObservable` as suggested):

1. using `args[0].EnsureObservableWrapped()`
2. using `args[0].WithAnnotation(ShouldBeObservableAnnotation.Instance)`